ui-tests/src/test/java/io/syndesis/qe/steps/customizations/extensions/ExtensionSteps.java

```diff
@@ -64,28 +64,86 @@ public void importExtension(DataTable extensionsData) throws Throwable {
+    public void importExtensionFromKnownFolder(DataTable extensionsData) throws Throwable {
+        CommonSteps commonSteps = new CommonSteps();
+
+        List<List<String>> dataTable = extensionsData.raw();
+
+        for (List<String> dataRow : dataTable) {
+
+            commonSteps.navigateTo("", NavigationElements.CUSTOMIZATIONS_NAV);
+            commonSteps.validatePage("", NavigationElements.CUSTOMIZATIONS_NAV);
+
+            commonSteps.clickOnLink(NavigationElements.EXTENSION_NAV);
+            commonSteps.validatePage("", NavigationElements.EXTENSION_NAV);
+
+            commonSteps.clickOnButton(NavigationElements.IMPORT_EXTENSION_BUTTON);
+            commonSteps.validatePage("", NavigationElements.IMPORT_EXTENSION_BUTTON);
+
+
+            uploadExtensionFromFile(dataRow.get(0));
+
+            commonSteps.clickOnButton(NavigationElements.IMPORT_EXTENSION_BUTTON);
+        }
+    }
+
+
+    @When("^.*uploads? extension \"([^\"]*)\"$")
+    public void uploadFile(String extensionName) throws Throwable {
+        uploadExtensionFromFile(extensionName);
+    }
+
     /**
-     * @deprecated use uploadExtensionFromFile method instead
+     * Uploads an extension from any relative path
+     *
+     * @param extensionName
+     * @param extensionPath
+     * @throws Throwable
      */
-    @Deprecated
-    @When("^Camilla upload extension \"([^\"]*)\"$")
-    public void uploadFile(String extensionName) throws Throwable {
-        //TODO temporary solution
-        String techExtensionFolderUrl = TestConfiguration.techExtensionUrl();
-        String techExtensionUrl = techExtensionFolderUrl + extensionName + ".jar";
+    @When("^.*uploads? extension with name \"([^\"]*)\" from relative path \"([^\"]*)\"$")
+    public void uploadExtensionFromFile(String extensionName, String extensionPath) throws Throwable {
+
+        String techExtensionUrl = extensionPath + extensionName;
         Path techExtensionJar = Paths.get(techExtensionUrl).toAbsolutePath();
         $(By.cssSelector("input[type='file']")).shouldBe(visible).uploadFile(techExtensionJar.toFile());
     }
 
-    @When("^Camilla upload extension with name \"([^\"]*)\" from relative path \"([^\"]*)\"$")
-    public void uploadExtensionFromFile(String extensionName, String extensionPath) throws Throwable {
 
-        String techExtensionUrl = extensionPath + extensionName;
+    /**
+     * Method will download extension .jar file from ../syndesis-extensions/extensionFolderName/target/*.jar
+     * You only have to specify folder name.
+     *
+     * @param extensionFolderName
+     * @throws Throwable
+     */
+    @When("^.*uploads? extension with name \"([^\"]*)\" from syndesis-extensions dir$")
+    public void uploadExtensionFromFile(String extensionFolderName) throws Throwable {
+        String defaultPath = "../syndesis-extensions/" + extensionFolderName + "/target/";
+
+        File[] files = new File(defaultPath).listFiles();
```

Comment:
You can apply filter directly via `listFiles()` method and then just assert for `size == 1`. As a bonus you can also add check if the default path exists first.

```java
listFiles((dir, name) -> !name.contains("original") && name.endsWith(".jar"));
Assertions.assertThat(files).hasSize(1).doesNotContainNull();
files[0].getName();
```